backend/core/src/main/java/org/sonarsource/sonarlint/core/BindingClueProvider.java

```diff
@@ -141,38 +170,57 @@ private Set<String> matchConnections(BindingClue bindingClue, Set<String> eligib
   }
 
   @CheckForNull
-  private static ScannerProperties extractScannerProperties(ClientFile matchedFile) {
-    LOG.debug("Extracting scanner properties from {}", matchedFile);
-    var properties = new Properties();
+  private static ConnectionProperties  extractSonarLintConfiguration(ClientFile sonarLintConfigurationFile) {
     try {
-      properties.load(new StringReader(matchedFile.getContent()));
-    } catch (Exception e) {
-      LOG.error("Unable to parse content of file '{}'", matchedFile, e);
+      var configuration = new Gson().fromJson(sonarLintConfigurationFile.getContent(), JsonObject.class);
+      var projectKey = configuration.get("projectKey");
+      var organization = configuration.get("organization");
+      var serverUrl = configuration.get("serverUrl");
+      return new ConnectionProperties(projectKey != null ? projectKey.getAsString() : null,
+        organization != null ? organization.getAsString() : null,
+        serverUrl != null ? serverUrl.getAsString() : null);
+    } catch (JsonSyntaxException e) {
```

Comment:
I wouldn't expect to log something here, since we fetch all `.json` files under `.sonarlint`, maybe in the future we'll have other files, and here the parsing is used to differentiate whether it's a SonarLint file for sharing connected mode or not. It could be possible and normal to fail here.